File: modules/angular2/test/router/spies.ts
import {Router, RouterOutlet, Location} from 'angular2/router';
import {SpyObject, proxy} from 'angular2/testing_internal';

export class SpyRouter extends SpyObject {
  constructor() { super(Router); }
}

export class SpyRouterOutlet extends SpyObject {
  constructor() { super(RouterOutlet); }
}

export class SpyLocation extends SpyObject {
  constructor() { super(Location); }
}

File: test/router/spies.dart
library router.spies;

import 'package:angular2/router.dart';
import 'package:angular2/testing_internal.dart';

@proxy
class SpyLocation extends SpyObject implements Location {
  noSuchMethod(m) => super.noSuchMethod(m);
}

@proxy
class SpyRouter extends SpyObject implements Router {
  noSuchMethod(m) => super.noSuchMethod(m);
}

@proxy
class SpyRouterOutlet extends SpyObject implements RouterOutlet {
  noSuchMethod(m) => super.noSuchMethod(m);
}
